src/SystemManager.cc

```diff
@@ -409,3 +426,94 @@ void SystemManager::ProcessPendingEntitySystems()
   }
   this->systemsToAdd.clear();
 }
+
+template <typename T>
+struct identity
+{
+    using type = T;
+};
+
+//////////////////////////////////////////////////
+/// TODO(arjo) - When we move to C++20 we can just use erase_if
+/// Removes all items that match the given predicate.
+/// This function runs in O(n) time and uses memory in-place
+template<typename Tp>
+void RemoveFromVectorIf(std::vector<Tp>& vec,
+  typename identity<std::function<bool(const Tp&)>>::type pred)
+{
+  auto originalSize = vec.size();
+  int j = 0;
+
+  for(std::size_t i = 0; i < vec.size(); ++i)
+  {
+    if (pred(vec[i]))
+    {
+      j++;
+    }
+    else
+    {
+      vec[i-j] = vec[i];
+    }
+  }
+
+  while (vec.size() > originalSize - j)
+  {
+    vec.pop_back();
+  }
```

Comment:
Can you use `std::vector::erase` and `std::remove_if`? https://stackoverflow.com/a/39019851/283225

Reply:
24f539dae